ucte/ucte-converter/src/main/java/com/powsybl/ucte/converter/UcteExporter.java

```diff
@@ -702,7 +702,7 @@ private static void setSwitchCurrentLimit(UcteLine ucteLine, Switch sw) {
         }
     }
 
-    private static UctePowerPlantType energySourceToUctePowerPlantType(EnergySource energySource) {
+    private static UctePowerPlantType energySourceToUctePowerPlantType(EnergySource energySource, String uctePowerPlantType) {
```

Comment:
Change the signature of this method to pass the generator instead of these two values.
Look for the property. If it's defined, use it. If the property is not defined, use the existing code without modifications.